x-pack/plugins/serverless_search/public/application/components/connectors/edit_service_type.tsx

```diff
@@ -62,24 +64,125 @@ export const EditServiceType: React.FC<EditServiceTypeProps> = ({ connector, isD
+      >
+        <EuiFlexGroup gutterSize="m">
+          <EuiFlexItem grow={false}>{_prepend}</EuiFlexItem>
+          <EuiFlexItem
+            grow={false}
+            data-test-subj={`serverlessSearchConnectorServiceType-${serviceType}`}
+          >
+            <EuiText size="s" textAlign="left">
+              {label}
+            </EuiText>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+        <EuiFlexItem grow={false}>{_append}</EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  };
+
   return (
-    <EuiFormRow
-      label={i18n.translate('xpack.serverlessSearch.connectors.serviceTypeLabel', {
-        defaultMessage: 'Connector type',
-      })}
-      data-test-subj="serverlessSearchEditConnectorType"
-      fullWidth
-    >
-      <EuiSuperSelect
-        // We only want to allow people to set the service type once to avoid weird conflicts
-        disabled={Boolean(connector.service_type) || isDisabled}
-        data-test-subj="serverlessSearchEditConnectorTypeChoices"
-        isLoading={isLoading}
-        onChange={(event) => mutate(event)}
-        options={options}
-        valueOfSelected={connector.service_type || undefined}
+    <>
+      <EuiFormRow
+        label={i18n.translate('xpack.serverlessSearch.connectors.serviceTypeLabel', {
+          defaultMessage: 'Connector type',
+        })}
+        data-test-subj="serverlessSearchEditConnectorType"
         fullWidth
-      />
-    </EuiFormRow>
+      >
+        <EuiComboBox
+          aria-label={i18n.translate(
+            'xpack.serverlessSearch.connectors.chooseConnectorSelectable.euiComboBox.accessibleScreenReaderLabelLabel',
+            { defaultMessage: 'Select a data source for your connector to use.' }
+          )}
+          // We only want to allow people to set the service type once to avoid weird conflicts
+          isDisabled={Boolean(connector.service_type) || isDisabled}
+          isLoading={isLoading}
+          data-test-subj="serverlessSearchEditConnectorTypeChoices"
+          prepend={<EuiIcon type={selectedConnector?.iconPath ?? connectorLogo} size="l" />}
+          singleSelection
+          fullWidth
+          placeholder={i18n.translate(
+            'xpack.serverlessSearch.connectors.chooseConnectorSelectable.placeholder.text',
+            { defaultMessage: 'Choose a data source' }
+          )}
+          options={initialOptions}
+          selectedOptions={selectedOption}
+          onChange={(selectedItem) => {
+            setSelectedOption(selectedItem);
+            if (selectedItem.length === 0) {
+              setSelectedConnector(null);
+              return;
+            }
+            const keySelected = Number(selectedItem[0].key);
+            setSelectedConnector({
+              ...connector,
+              ...allConnectors[keySelected],
+            });
+            mutate(allConnectors[keySelected].serviceType);
+          }}
+          renderOption={renderOption}
+          rowHeight={(euiTheme.base / 2) * 5}
```

Comment:
nitpick, none of the stock sizes are close to what we need?

Reply:
Not really seems I tried back and forth but this is the most accurate value we should use